src/lib.rs

```diff
@@ -211,6 +223,60 @@ impl<const N: usize> Bytes<N> {
     }
 }
 
+/// Construct a `Bytes<N>` instance from an array with `N` elements.
+///
+/// Currently, the array is copied, but a more efficient implementation could be used in the
+/// future.
```

Comment:
Is it really copied? I think this only depends on the compiler optimizations being smart enough to avoid moving around memory, but that would also apply if we just used `transmute` for example. This depends on the stack layout so I'm not sure it's worth mentioning.

Reply:
Admittedly I did not look at the generated code but I think we saw previously that `Vec::extend_from_slice` is not optimized well so I would expect a transmute to do better.  But I can also just drop that line if you prefer.

Reply:
Nah, let's just merge.